### PR TITLE
Better error handling and environment clearing

### PR DESCRIPTION
**Pull request by antho1404**
Fix https://github.com/liteflow-labs/liteflow-js/issues/12

Now any exit of the CLI on the commands that starts the engine will stop it whatever the reason of the exit, error, normal exit, process killed...

Fo that I had to do a few things:
- Don't return any state from the first steps to execute
- All data from the commands are stored as attribute of the class
- Created a `onExit` function that is attached to any process exit from the beginning (init function)

### CODE REVIEW

File: packages/cli/src/commands/process/dev.ts [hunk 1]
```diff
@@ -43,127 +45,143 @@ export default class Dev extends Command {
...
+          return this.log(prefix +
+            '\n\tinputs:  ' + chalk.gray(JSON.stringify(decode(execution.inputs || {}))) +
+            '\n\toutputs: ' + chalk.gray(JSON.stringify(decode(execution.outputs || {}))) +
+            '\n')
+        })
+    } catch (e) { }
+  }
+
+  async onExit() {
+    await new Listr<Environment.IStop>([
```
Comment (NicolasMahe) [inline packages/cli/src/commands/process/dev.ts:161]: each following task should also have the error catched.
I got the following error and the engine is still running:
```
➜  app ../liteflow-js/packages/cli/bin/run process:dev ./process-test.yml --version=dev
  ✔ Starting environment
  ✔ Compiling process
  ✔ Creating services
  ❯ Starting services
    ✖ 84P85npNWfUFizjkCpvQVxs1Nj291EC9MGiQEeMdkGZ7
      → (HTTP code 500) server error - {"message":"Cannot locate specified Dockerfile: Dockerfile"}
      BPktXrtNTYEmsZR4HkN1PtwBKUd15KXuXMx7kiGwEgM1
    Creating process
    Fetching process's logs
  ↓ Stopping logs [skipped]
  ↓ Deleting process [skipped]
  ✖ Stopping services
    → invalid request: runner "84P85npNWfUFizjkCpvQVxs1Nj291EC9MGiQEeMdkGZ7" not found
    Stopping environment
```

Comment (NicolasMahe) [reply packages/cli/src/commands/process/dev.ts:161]: same comment apply to all dev commands

Comment (antho1404) [reply packages/cli/src/commands/process/dev.ts:161]: fixes done, I just had better handling of the data by testing the existence of them before delete

**Comment by antho1404** [comment]
all fixes are now done. Should be good to go

**Comment by NicolasMahe** [comment]
It's much better but still not perfect. I still have the environment not clean up when services cannot be deleted because maybe they are used by another process still running.
```
➜  liteflow-js git:(fix/error-handling-clear-env) liteflow-dev dev ../app/
  ✔ Starting environment
  ✔ Compiling processes
  ✔ Creating services
  ✔ Starting services
  ✔ Creating processes
  ✔ Fetching logs
^C ›   Warning: Result stream error: 1 CANCELLED: Cancelled
  ✔ Stopping logs
  ❯ Stopping running services
    ⠦ A4chyLLJ2ZcPmwM4iKPcMBBgPT7cbWhX9GewUmtkfZzB
      A35GdNmQXifCq1Upvexr3asvvGN6wz7ytGefPogVUTM5
    Deleting processes
    Stopping environment
```
----
I would suggest removing the "multi-environment" functionality in order to simplify the logic and ALWAYS clean up everything at the end of the command.

**Comment by antho1404** [comment]
Issue added based on your feedback https://github.com/liteflow-labs/liteflow-js/issues/52

**Pull request merged**